sklearn/preprocessing/label.py

```diff
@@ -138,12 +189,23 @@ def fit(self, X, y, sample_weight=None):
         self : object
             Returns self.
         """
+        # memorize if y should be a multi-output
+        self.y_shape_ = y.shape
```

Comment:
hm maybe we shouldn't remember ``y.shape[0]``? Though dummy tries that and then has two flags ``n_outputs_`` and ``outputs_2d_``.